tests/test_panelinha_1.py

```diff
@@ -10,26 +12,30 @@ class TestPanelinhaScraper(ScraperTest):
     def test_host(self):
         self.assertEqual("panelinha.com.br", self.harvester_class.host())
 
+    @unittest.skip("canonical_url is not available from this webpage")
+    def test_canonical_url(self):
+        self.assertEqual(
+            "https://panelinha.com.br/receita/rosbife",
+            self.harvester_class.canonical_url(),
+        )
+
     def test_title(self):
         self.assertEqual(self.harvester_class.title(), "Rosbife")
 
     def test_author(self):
         self.assertEqual(self.harvester_class.author(), "Panelinha")
 
-    def test_total_time(self):
-        self.assertEqual(60, self.harvester_class.total_time())
-
     def test_yields(self):
         self.assertEqual("4 servings", self.harvester_class.yields())
 
     def test_ingredients(self):
         self.assertEqual(
             [
-                "1 peça de filé mignon para rosbife (cerca de 750 g)",
-                "1 colher (chá) de mostarda amarela em pó",
-                "1 colher (chá) de páprica defumada",
+                "750 g de filÃ© mignon em peÃ§a para rosbife",
```

Comment:
Arg.  I've noticed while doing a final readthrough that there's a problem with the text decoding here.  I'm taking a bit of a look at that now.

Reply:
[`charchef`](https://pypi.org/project/charchef/) does a good job of fixing these malencoded text entries, but is fairly slow the first time it runs:

```pyi
>>> from charchef import aa_convert_utf8_to_ascii_,aa_repair_bad_conversion_to_utf8,aa_replace_non_printable_chars
>>> text = '1 colher (chÃ¡) de pÃ¡prica defumada'
>>> aa_repair_bad_conversion_to_utf8(text)
['1 colher (chá) de páprica defumada']
```

(that took 10 seconds or so, maybe slightly longer)

Reply:
[`LatinFixer`](https://github.com/hansalemaos/LatinFixer) by the same author can achieve the same result here, and fast!

```pyi
>>> from LatinFixer import LatinFix
>>> text = '1 colher (chÃ¡) de pÃ¡prica defumada'
>>> LatinFix(text).apply_wrong_chars()
1 colher (chá) de páprica defumada
```

Reply:
I've pushed a possible change to do this; it isn't perfect yet, though.  Some ingredient lines begin with fractional amounts -- `⅔`, for example -- and those aren't being decoded clearly yet.

@jknndy do you remember what browser you downloaded the `testhtml` files with?  There's a possibility that something in the encoding headers/negotiation that it used at the time has resulted in unusual results in the downloaded HTML.  When I view the original recipe page live in a web browser, I find that it's more readable than when I view the corresponding `.testhtml` file locally in the same browser.

Reply:
I've resolved this problem by re-downloading a copy of the HTML; the text encoding received from the site and/or the encoding used to write the file seem to have resolved the problem.  So despite that `LatinFixer` adventure (it's useful to know about that library), it isn't added as a dependency/string handler here at the moment.

Reply:
I grabbed all the updated test html using the header defined in _abstract so it's strange that would have caused an issue. Once I'm down with the last batch I have a few to revisit that I skipped over, I'll add this one to this list & try some other recipes to see what happens.